builtin/mainmenu/tab_client_mods.lua

```diff
@@ -0,0 +1,128 @@
+--Minetest
+--Copyright (C) 2013 sapier
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+--------------------------------------------------------------------------------
+
+function get_client_mods()
+    local path = core.get_clientmodpath()
+    local client_mods = core.get_dir_list(path, true)
+    local retval = {}
+    for _, name in ipairs(client_mods) do
+        if name:sub(1, 1) ~= "." then
+            table.insert(retval,name)
+        end
+    end
+    return(retval)
+end
+
+local function get_mods_conf()
+    local client_modpath = core.get_clientmodpath()
+
+    local f = io.open(client_modpath..DIR_DELIM.."mods.conf", "r")
+    local temp_settings = {}
+
+    if f then
+        for line in io.lines(client_modpath..DIR_DELIM.."mods.conf") do
```

Comment:
I don't know if this just copied code, but this should definitely use a [`Settings` object](https://github.com/minetest/minetest/blob/0a1181f7637adab6fe1ba5d3c09038912a06b37f/doc/lua_api.txt#L6436) instead

Reply:
No, this is designed to remove old data, and using a Settings object does not remove old data automatically. What is happening is that the mods.conf settings file is getting rewritten completely dynamically.

Reply:
But you could read the file using a settings object and still write it the old way.